### PR TITLE
Add tests for creator-staking pallet

### PR DESCRIPTION
**Pull request by F3Joule**

### CODE REVIEW

**Review by olehmell — CHANGES_REQUESTED** [review]

File: pallets/creator-staking/src/tests/tests.rs [hunk 1]
```diff
@@ -0,0 +1,2077 @@
...
+
+        for _ in 1..CreatorStaking::current_era() {
+            assert_noop!(
+                CreatorStaking::withdraw_from_unregistered(
+                    RuntimeOrigin::signed(backer),
+                    creator_id
+                ),
+                Error::<TestRuntime>::UnclaimedRewardsRemaining
+            );
+            assert_claim_backer(backer, creator_id, false);
```
Comment (olehmell) [inline pallets/creator-staking/src/tests/tests.rs:621]: What is the user trying to claim w/ restake?

File: pallets/creator-staking/src/tests/tests.rs [hunk 2]
```diff
@@ -0,0 +1,2077 @@
...
+            Balances::free_balance(&backer_id) - BackerLocksByAccount::<TestRuntime>::get(backer_id).total_locked;
+        assert_eq!(MINIMUM_REMAINING_AMOUNT, transferable_balance);
+
+        // stake some amount, a bit less than free balance
+        let backer_id = 3;
+        let backer_free_balance =
+            Balances::free_balance(&backer_id).saturating_sub(MINIMUM_REMAINING_AMOUNT);
+        assert_stake(backer_id, creator_id, backer_free_balance - 200);
+
+        // Try to stake more than we have available (since we already locked most of the free balance).
```
Comment (olehmell) [inline pallets/creator-staking/src/tests/tests.rs:732]: Maybe sense to explain why we dont expect error

File: pallets/creator-staking/src/tests/tests.rs [hunk 3]
```diff
@@ -0,0 +1,2077 @@
...
+        assert_stake(backer_id, creator_id, 500);
+    })
+}
+
+#[test]
+fn stake_on_unregistered_creator_fails() {
```
Comment (olehmell) [inline pallets/creator-staking/src/tests/tests.rs:738]: unregistered creator, but an error is Inactive

File: pallets/creator-staking/src/tests/tests.rs [hunk 4]
```diff
@@ -0,0 +1,2077 @@
...
+
+#[test]
+fn unstake_with_zero_value_is_not_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let creator_id = 1;
+        assert_register(10, creator_id);
+
+        assert_noop!(
```
Comment (olehmell) [inline pallets/creator-staking/src/tests/tests.rs:946]: Maybe make sense to stake somethign before unstake 0

File: pallets/creator-staking/src/tests/tests.rs [hunk 5]
```diff
@@ -0,0 +1,2077 @@
...
+            CreatorStaking::unstake(RuntimeOrigin::signed(1), creator_id, 10),
+            Error::<TestRuntime>::NotStakedCreator,
+        );
+    })
+}
+
+#[test]
+fn unstake_too_many_era_stakes() {
```
Comment (olehmell) [inline pallets/creator-staking/src/tests/tests.rs:1023]: I totally didnt understand the test `unstake_too_many_era_stakes`, we are doing `MAX_ERA_STAKE_VALUES` stakes and after this try to do `unstake` and expect error `TooManyEraStakeValues`, why?

File: pallets/creator-staking/src/tests/tests.rs [hunk 6]
```diff
@@ -0,0 +1,2077 @@
...
+        assert_noop!(
+            CreatorStaking::unstake(RuntimeOrigin::signed(backer_id), creator_id, 10),
+            Error::<TestRuntime>::TooManyEraStakeValues
+        );
+    })
+}
+
+#[ignore]
+#[test]
+fn unstake_with_no_chunks_allowed() {
```
Comment (olehmell) [inline pallets/creator-staking/src/tests/tests.rs:1049]: Need to clarification in meet

File: pallets/creator-staking/src/tests/tests.rs [hunk 7]
```diff
@@ -0,0 +1,2077 @@
...
+        let backer = 2;
+        let creator_id = 1;
+
+        // stake some tokens
+        let start_era = CreatorStaking::current_era();
+        assert_register(stakeholder, creator_id);
+        let stake_value = 100;
+        assert_stake(backer, creator_id, stake_value);
+
+        // disable reward restaking
```
Comment (olehmell) [inline pallets/creator-staking/src/tests/tests.rs:1462]: I guess it isnt correct comment for the line

File: pallets/creator-staking/src/tests/tests.rs [hunk 8]
```diff
@@ -0,0 +1,2077 @@
...
+
+        // unstake all the tokens
+        assert_unstake(backer, creator_id, stake_value);
+
+        // ensure claimed value goes to claimer's free balance
+        assert_claim_backer(backer, creator_id, false);
+    })
+}
+
+// FIXME: how to be here?
```
Comment (olehmell) [inline pallets/creator-staking/src/tests/tests.rs:1494]: So, are you going to fix?

File: pallets/creator-staking/src/tests/tests.rs [hunk 9]
```diff
@@ -0,0 +1,2077 @@
...
+        assert_register(stakeholder, creator_id);
+
+        // Stake with MAX_ERA_STAKE_VALUES - 1 on the same creator. It must work.
+        let start_era = CreatorStaking::current_era();
+        for offset in 1..MAX_ERA_STAKE_VALUES {
+            assert_stake(backer_id, creator_id, 100);
+            advance_to_era(start_era + offset * 5);
+        }
+
+        // claim and restake once, so there's a claim record for the for the current era in the stakes vec
```
Comment (olehmell) [inline pallets/creator-staking/src/tests/tests.rs:1513]:
```suggestion
        // claim and restake once, so there's a claim record for the current era in the stakes vec
```

File: pallets/creator-staking/src/tests/tests.rs [hunk 10]
```diff
@@ -0,0 +1,2077 @@
...
+    })
+}
+
+// Inflation tests
+// ---------------
+
+#[test]
+fn default_reward_distribution_config_is_consitent() {
+    let reward_config = RewardsDistributionConfig::default();
+    assert!(reward_config.is_consistent());
```
Comment (siman) [inline pallets/creator-staking/src/tests/tests.rs:1761]: Rename `is_consistent` to `is_sum_equal_to_one`?

File: pallets/creator-staking/src/tests/tests.rs [hunk 11]
```diff
@@ -0,0 +1,2077 @@
...
+#[test]
+pub fn set_configuration_fails() {
+    ExternalityBuilder::build().execute_with(|| {
+        // 1
+        assert_noop!(
+            CreatorStaking::set_reward_distribution_config(RuntimeOrigin::signed(1), Default::default()),
+            BadOrigin
+        );
+
+        // 2
```
Comment (siman) [inline pallets/creator-staking/src/tests/tests.rs:1831]:
```suggestion
        // Root cannot set invalid config
```

File: pallets/creator-staking/src/lib.rs [hunk 12]
```diff
@@ -230,6 +227,7 @@ pub mod pallet {
         CreatorUnregisteredWithSlash { creator_id: CreatorId, slash_amount: BalanceOf<T> },
         NewCreatorStakingEra { number: EraIndex },
         MaintenanceModeSet { enabled: bool },
+        ActiveDistributionConfigurationChanged { new_config: RewardsDistributionConfig },
```
Comment (siman) [inline pallets/creator-staking/src/lib.rs:230]:
```suggestion
        RewardsDistributionConfigChanged { new_config: RewardsDistributionConfig },
```

File: pallets/creator-staking/src/tests/tests.rs (hunk 6 above)
Comment (F3Joule) [reply pallets/creator-staking/src/tests/tests.rs:1049]: There is a comment right on the next line :)
Ok, will move it.

**Review by siman — CHANGES_REQUESTED** [review]

**Review by siman — APPROVED** [review]

**Pull request merged**